TodoList/Tests/TodoList.Api.Tests/Controllers/ListItemsControllerTest.cs

```diff
@@ -1,166 +1,178 @@
+        [TestCase("    ")]
+        [TestCase(null)]
+        public void PostAsync_ItemWithNoText_ReturnsCorrectErrorResponse(string postedText)
+        {
+            var expectedKeys = new[] { "Text" };
+            var postedListItem = new ListItem { Id = Guid.Empty, Text = postedText };
+
+            var actionResult = _controller.PostAsync(postedListItem).Result;
+            var responseMessage = actionResult.ExecuteAsync(CancellationToken.None).Result;
+            HttpError error;
+            responseMessage.TryGetContentValue(out error);
+            var actualLocation = responseMessage.Headers.Location;
+
+            Assert.That(responseMessage.StatusCode, Is.EqualTo(HttpStatusCode.BadRequest));
+            Assert.That(error.ModelState.Keys, Is.EqualTo(expectedKeys));
+            Assert.That(actualLocation, Is.EqualTo(null));
+        }
+
+        [Test]
+        public void PostAsync_ItemWithNonEmptyGuid_ReturnsCorrectErrorResponse()
+        {
+            var expectedKeys = new[] { "Id" };
+            var postedListItem = new ListItem { Id = _guidOfFirstItem, Text = "text" };
+
+            var actionResult = _controller.PostAsync(postedListItem).Result;
+            var responseMessage = actionResult.ExecuteAsync(CancellationToken.None).Result;
+            HttpError error;
+            responseMessage.TryGetContentValue(out error);
+            var actualLocation = responseMessage.Headers.Location;
+
+            Assert.That(responseMessage.StatusCode, Is.EqualTo(HttpStatusCode.BadRequest));
+            Assert.That(error.ModelState.Keys, Is.EqualTo(expectedKeys));
+            Assert.That(actualLocation, Is.EqualTo(null));
+        }
+
+        [Test]
+        public void PostAsync_WithNullArguments_ReturnsCorrectErrorResponse()
+        {
+            var expectedKeys = new[] { string.Empty };
+
+            var actionResult = _controller.PostAsync(null).Result;
+            var responseMessage = actionResult.ExecuteAsync(CancellationToken.None).Result;
+            HttpError error;
+            responseMessage.TryGetContentValue(out error);
+            var actualLocation = responseMessage.Headers.Location;
+
+            Assert.That(responseMessage.StatusCode, Is.EqualTo(HttpStatusCode.BadRequest));
+            Assert.That(error.ModelState.Keys, Is.EqualTo(expectedKeys));
+            Assert.That(actualLocation, Is.EqualTo(null));
+        }
+
+        [Test]
+        public void PostAsync_WithValidArguments_ReturnsCorrectResponse()
+        {
+            var expectedListItem = new ListItem { Id = _guidOfFirstItem, Text = "text" };
+            var postedListItem = new ListItem { Id = Guid.Empty, Text = "newText" };
+            var expectedLocation = new Uri($"api/v1/ListItems/{_guidOfFirstItem}", UriKind.Relative);
+            _repository.CreateAsync(postedListItem).Returns(expectedListItem);
+            _urlGenerator.GenerateUrl(expectedListItem).Returns($"api/v1/ListItems/{expectedListItem.Id}");
```

Comment:
You should actually test whether correct item got passed to the method. Compose the substitute in a way it reads id from the argument passed to the method (you can leverage the [`callInfo` that NSubstitute provides](http://nsubstitute.github.io/help/return-from-function/))